app/src/main/java/com/raghav/spacedawn/ui/fragments/ArticlesListFragment.kt

```diff
@@ -7,31 +7,68 @@ import android.widget.AbsListView
 import android.widget.Toast
 import androidx.fragment.app.Fragment
 import androidx.fragment.app.viewModels
-import androidx.lifecycle.Observer
+import androidx.lifecycle.lifecycleScope
 import androidx.navigation.fragment.findNavController
 import androidx.recyclerview.widget.LinearLayoutManager
 import androidx.recyclerview.widget.RecyclerView
 import com.raghav.spacedawn.R
 import com.raghav.spacedawn.adapters.ArticlesAdapter
 import com.raghav.spacedawn.databinding.FragmentArticlesListBinding
-import com.raghav.spacedawn.ui.AppViewModel
+import com.raghav.spacedawn.ui.viewmodels.ArticlesListFragmentVM
 import com.raghav.spacedawn.utils.Constants.Companion.QUERY_PAGE_SIZE
 import com.raghav.spacedawn.utils.Resource
 import dagger.hilt.android.AndroidEntryPoint
 
 @AndroidEntryPoint
 class ArticlesListFragment : Fragment(R.layout.fragment_articles_list) {
 
-    private val viewModel by viewModels<AppViewModel>()
-    lateinit var articlesAdapter: ArticlesAdapter
+    private val viewModel by viewModels<ArticlesListFragmentVM>()
+    private lateinit var articlesAdapter: ArticlesAdapter
     private lateinit var binding: FragmentArticlesListBinding
-    private val TAG = "ArticlesListFragment"
+
+    companion object {
+        const val TAG = "ArticlesListFragment"
+    }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
         binding = FragmentArticlesListBinding.bind(view)
         setupRecyclerView()
 
+        viewLifecycleOwner.lifecycleScope.launchWhenStarted {
+            viewModel.articlesFlow.collect {
+                Log.d(TAG, "ui-collector-triggered for $it")
```

Comment:
logs for debugging are usually avoided if a log is needed i prefer to use `i`

Reply:
will remove the debugging logs